doc/api/module.md

```diff
@@ -80,15 +80,27 @@ isBuiltin('fs'); // true
 isBuiltin('wss'); // false
 ```
 
-### `module.register()`
+### `module.register(specifier[, parentURL[, options]])`
```

Comment:
```suggestion
### `module.register(specifier[, parentURL][, options])`
```

Reply:
I copied the style from https://nodejs.org/api/fs.html#filehandlewritestring-position-encoding

Reply:
Actually in the `node:fs ` docs, the same function is listed several times when it accepts more than one signature, I don’t know if it would make sense here.

Reply:
Do you see any examples of the pattern you’re suggesting? I see ones like `fs.openSync(path[, flags[, mode]])` or  `fs.writeSync(fd, buffer, offset[, length[, position]])`. I think what I have here is correct at least per what I see in `fs.md`.

Reply:
indeed, if parentURL is required when options is provided, the current bracket style is necessary - this change implies you can omit the second arg and pass the third and it will work.

Reply:
What Jordan said, you can omit the `parentURL`, my suggestion conveys that in a more compact way than the `node:fs` docs.

> Do you see any examples of the pattern you’re suggesting?

There are lots, the first one I found: https://github.com/nodejs/node/blob/6bef80a50b5308c681ff214fc1a38217c40390e5/doc/api/fs.md#L367 https://github.com/nodejs/node/blob/6bef80a50b5308c681ff214fc1a38217c40390e5/doc/api/fs.md#L391 https://github.com/nodejs/node/blob/6bef80a50b5308c681ff214fc1a38217c40390e5/doc/api/fs.md#L420

Reply:
wait, so you can pass all of (specifier, options), (specifier, parentURL), (specifier), or (specifier, parentURL, options)? That’s a confusing API; if parentURL is optional, it should be in options.

Reply:
> so you can pass all of (specifier, options), (specifier, parentURL), (specifier), or (specifier, parentURL, options)?

That's correct

> if parentURL is optional, it should be in options.

You can put it in options.

Reply:
Then it’s very confusing that you aren’t required to put it in options to pass it :-/

but given this API, the suggested change is correct.

Reply:
Well if you compare to `filehandle.read`, it's about the same (you can pass `buffer` in the `options` or as a standalone argument), yet I don't think many folks find it confusing.

Reply:
It certainly matches early node APIs, and userland APIs from that time, but i was under the impression that the ecosystem consensus was to stop making these kind of overloaded APIs because they’re confusing.